cmd/share-list-folders.go

```diff
@@ -30,21 +30,34 @@ func shareListFolders(cmd *cobra.Command, args []string) (err error) {
 		return
 	}
 
-	// TODO(bonafidehan): handle paging. Currently uses default limit of 1000.
+	printFolders(res.Entries)
 
-	for _, f := range res.Entries {
-		fmt.Printf("%v\t%v\n", f.PathLower, f.PreviewUrl)
+	for len(res.Cursor) > 0 {
+		continueArg := sharing.NewListFoldersContinueArg(res.Cursor)
+
+		res, err = dbx.ListFoldersContinue(continueArg)
+		if err != nil {
+			return
+		}
+
+		printFolders(res.Entries)
```

Comment:
It does iterate until finished, since `res` is being set anew each time. Tested already by setting a very small `limit` and observing it lists everything.